snowplow-demo-app/src/main/java/com/snowplowanalytics/snowplowtrackerdemo/Demo.java

```diff
@@ -206,17 +213,19 @@ private void setupAndTrackDemoEvents() {
         Protocol security = _security.getCheckedRadioButtonId() ==
                 _radioHttp.getId() ? Protocol.HTTP : Protocol.HTTPS;
 
-        if (!e.getEmitterStatus()) {
-            e.setEmitterUri(uri);
-            e.setRequestSecurity(security);
-            e.setHttpMethod(method);
+        EmitterController e = Snowplow.getDefaultTracker().getEmitter();
```

Comment:
I really quite like this default tracker stuff. I was thinking about implementing diagnostic events in the JS tracker but it's hard without the default tracker concept. I _might_ be stealing this ;)

Reply:
Interesting, about the diagnostic there are some parts I would do differently now. I should probably put some of those ideas/feedbacks in a shared doc to discuss this more widely.